### PR TITLE
RHS Compat ADD Missing (undefined) Weapons

### PR DESCRIPTION
**Pull request by Lazejun**
As Written in Issue #7560 there were Weapons missing in the Compats.
I also took the Time to look into the Comming RHS version (0.51).

**Added:**
**RHS USAF:**
~~I used "call ace_scopes_fnc_getBoreHeight" to get the Railabove bore values but i am not 100% sure they are correct.~~

Scar H CQC
ACE_barrelTwist =  304.8; // 1:12"
ACE_barrelLength = 330.0;

Scar H STD
ACE_barrelTwist =  304.8; // 1:12"
ACE_barrelLength = 406.0;

Scar H LB
ACE_barrelTwist =  304.8; // 1:12"
ACE_barrelLength = 508.0;

M14 Base       ----> New Parent for all Mk14s
ACE_barrelTwist = 304.8;
ACE_barrelLength = 558.8;

Soccom 16
ACE_barrelTwist = 304.8;
ACE_barrelLength = 413; //16.25 in (413 mm)

**Redefined**
M240 and M249, M249 PIP and M249 were not completly accepting Values, M240B was defined, M240G wasnt.

**RHS GREF:**
**Added**

VHSD2
ACE_barrelTwist= 177.8;
ACE_barrelLength=500.0;

VHSK2
ACE_barrelTwist= 177.8;
ACE_barrelLength=410.0;

VHSD2_CT15x
ACE_barrelTwist= 177.8;
ACE_barrelLength=500.0;

VHSD2_BG (GL Variant)
ACE_barrelTwist= 177.8;
ACE_barrelLength=500.0;

L1A1 Polymer/Wood
ACE_barrelTwist = 302.26;
ACE_barrelLength =554.4;

MG42
ACE_barrelTwist = 305;
ACE_barrelLength =530;

MP44
ACE_barrelTwist = 240;
ACE_barrelLength =420;

M3A1
ACE_barrelTwist = 406;
ACE_barrelLength =203.2;

M1 Garand
ACE_barrelTwist = 254;
ACE_barrelLength =610;

### CODE REVIEW

File: optionals/compat_rhs_gref3/CfgWeapons.hpp [hunk 1]
```diff
@@ -77,8 +77,44 @@ class CfgWeapons {
         ACE_barrelTwist=240;
         ACE_barrelLength=658;
     };
+    class rhs_weap_vhs2_base;
+    class rhs_weap_vhsd2: rhs_weap_vhs2_base {
+        ACE_barrelTwist= 177.8;
+        ACE_barrelLength=500.0;
```
Comment (commy2) [inline optionals/compat_rhs_gref3/CfgWeapons.hpp:83]: v

File: optionals/compat_rhs_usf3/CfgWeapons.hpp [hunk 2]
```diff
@@ -74,37 +74,70 @@ class CfgWeapons {
...
     };
     class rhs_weap_sr25_ec: rhs_weap_sr25 {
         ACE_barrelTwist = 285.75;
         ACE_barrelLength = 508.0;
-        ACE_RailHeightAboveBore = 3.13689;
+        ACE_RailHeightAboveBore = 1.98812;
+    };
+    class rhs_weap_SCAR_H_Base;
+    class rhs_weap_SCAR_H_CQC_Base : rhs_weap_SCAR_H_Base {
+        ACE_barrelTwist =  304.8; // 1:12"
```
Comment (commy2) [inline optionals/compat_rhs_usf3/CfgWeapons.hpp:128]: Please only exactly one space before and after the equal sign.

Comment (Lazejun) [reply optionals/compat_rhs_usf3/CfgWeapons.hpp]: Changed all Entries to Space Equal Space ( = )

**Comment by Lazejun** [comment]
Resolved the Conflict, adjusted how the M14s were defined (0.51 RHS USAF)

**Comment by dedmen** [comment]
> I used "call ace_scopes_fnc_getBoreHeight" to get the Railabove bore values but i am not 100% sure they are correct.

Thats not how that works, the scopes function returns the height from config, and it also doesn't return heightAboveBore, it returns RailHeightAboveBore+ScopeHeightAboveRail.

The config entry is there because the height cannot be determined using scripts and needs to be manually entered, so you cannot just run some script and enter the value it gives you.

**Comment by Lazejun** [comment]
Reverted Railabovebore to the Values that existed before and removed the ones i added.

**Pull request merged**